accounts/abi/bind/precompiletemplate.go

```diff
@@ -0,0 +1,315 @@
+	Structs  map[string]*tmplStruct  // Contract struct type definitions
+}
+
+// tmplPrecompileContract contains the data needed to generate an individual contract binding.
+type tmplPrecompileContract struct {
+	*tmplContract
+	AllowList bool                   // Indicator whether the contract uses AllowList precompile
+	Funcs     map[string]*tmplMethod // Contract functions that include both Calls + Transacts in tmplContract
+}
+
+// tmplSourcePrecompileGo is the Go precompiled source template.
+const tmplSourcePrecompileGo = `
+// Code generated
+// This file is a generated precompile contract with stubbed abstract functions.
+// The file is generated by a template. Please inspect every code and comment in this file before use.
+
+// There are some must-be-done changes waiting in the file. These are highlighted by CAPITAL WORDS.
+// Additionally there are other files you need to edit to activate your precompile.
+// These areas are highlighted with comments "ADD YOUR PRECOMPILE HERE".
+// For testing take a look at other precompile tests in core/stateful_precompile_test.go
+
+/* General guidelines for precompile development:
+1- Read the comment and set a suitable contract address in precompile/params.go. E.g:
+	{{.Contract.Type}}Address = common.HexToAddress("ASUITABLEHEXADDRESS")
+2- Set gas costs here
+3- Unless you know what you're doing write your custom code only after highlighted areas with "CUSTOM CODE STARTS HERE".
+Typically, custom codes are required in only those areas.
+4- Add your precompile upgrade in params/config.go
+5- Add your upgradable config in params/precompile_config.go
+6- Add your solidity interface and test contract to contract-examples/contracts
+7- Write solidity tests for your precompile in contract-examples/test
+8- Create e2e test for your solidity test in tests/e2e/solidity/suites.go
+9- Create your genesis with your precompile enabled in tests/e2e/genesis/
+10- Run your e2e precompile Solidity tests with 'E2E=true ./scripts/run.sh'
+
+*/
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	{{- range .Contract.Funcs}}
+	{{.Normalized.Name}}GasCost uint64 = -1 // SET A GAS COST HERE
+	{{- end}}
+
+	// {{.Contract.Type}}RawABI contains the raw ABI of {{.Contract.Type}} contract.
+	{{.Contract.Type}}RawABI = "{{.Contract.InputABI}}"
+)
+
+// Singleton StatefulPrecompiledContract and signatures.
+var (
+	_ StatefulPrecompileConfig = &{{.Contract.Type}}Config{}
+
+	{{.Contract.Type}}Precompile StatefulPrecompiledContract = create{{.Contract.Type}}Precompile({{.Contract.Type}}Address)
+	{{- range .Contract.Funcs}}
+	{{decapitalise .Normalized.Name}}Signature = CalculateFunctionSelector("{{.Original.Sig}}")
+	{{- if not .Original.IsConstant}}
+
+	ErrCannot{{.Normalized.Name}} = errors.New("non-enabled cannot {{.Original.Name}}")
```

Comment:
Where are these errors meant to be used?

Reply:
for allowlist check to return an error